src/main/app/logging/apiLogger.ts

```diff
@@ -25,7 +31,10 @@ export class ApiLogger {
       message: `API: Response ${responseData.responseCode} from ${responseData.uri} ` +
       ((responseData.responseBody && this.isDebugLevel()) ? `| Body: ${this._stringifyObject(responseData.responseBody)} ` : '') +
       ((responseData.error) ? `| Error: ${this._stringifyObject(responseData.error)} ` : ''),
-      responseCode: responseData.responseCode
+      responseCode: responseData.responseCode,
+      requestId: (responseData.requestHeaders) ? responseData.requestHeaders['Request-Id'] : '',
```

Comment:
shouldn't these always be set by the middleware?

Reply:
They should, but I was in a copy-pasting spree at that moment.